torch/csrc/distributed/rpc/tensorpipe_utils.h

```diff
@@ -12,6 +20,87 @@ namespace torch {
 namespace distributed {
 namespace rpc {
 
+#ifdef USE_CUDA_NOT_ROCM
+using at::cuda::CUDAStream;
+#endif
+
+
+// A general device context class for both CPU and CUDA. If CUDA is not
+// available, all CUDA-related methods will be no-ops.
+struct FullDeviceContext {
+
+  FullDeviceContext(const FullDeviceContext& other) = delete;
+  FullDeviceContext(FullDeviceContext&& other) = delete;
+
+  FullDeviceContext& operator=(const FullDeviceContext& rhs) = delete;
+  FullDeviceContext& operator=(FullDeviceContext&& rhs) & = delete;
+
+  explicit FullDeviceContext(bool /* unused */) {}
+  virtual void recordDataPtrs(
+      const std::vector<c10::DataPtr>& dataPtrs) const {}
+  virtual void recordTensors(const std::vector<torch::Tensor>& tensors) const {}
+  virtual void blockCurrentStreams() const {}
+  virtual void waitForCurrentStreams() const {}
+  virtual void synchronize() const {}
+
+#ifdef USE_CUDA_NOT_ROCM
```

Comment:
If we use `c10::Stream` here, we will need to introduce `c10::Stream` -> `CUDAStream` conversions in multiple places, e.g.,  when we call `CUDAEvent.block` or `CUDAEvent.record`, or when we use `CUDAMultiStreamGuard`, etc. Let me know if you prefer adding conversions at callsite instead of using ifdef here.

Reply:
I'll leave to you the decision between the `#ifdef`s and the conversions.